Remove warnings and errors that occur when using GCC >= 7.0.

allenh1 opened the pull request:
This is another small GCC >= 7 change.

connects to ros2/rclcpp#343

Commit:
recompute cluster stat when force_publication

When force_publication is True, and no resample has been done the statistic
about the current clusters of particles are not recomputed. So amcl published
an outdated information.

This commit introduces a call to `pf_cluster_stats` when required (i.e no
resample but force_publication).

selvasamuel committed:
Fixed deadlock when changing global planner

azaganidis committed:
Initialization of filter with GPS and odometry.

wjwwood committed:
disable everything for now with AMENT_IGNORE files

wjwwood committed:
fix cmake use of yaml-cpp and sdl / sdl-image

wjwwood committed:
remove trailing whitespace from map_server package (#557)

wjwwood committed:
ROS 2 port of map_server (#1)

* remove AMENT_IGNORE and vestigial files

* style fixes, trailing whitespace and typos

* port to ROS 2

* use existing rclcpp syntax for now

wjwwood committed:
remove duplicate and unreferenced file (#558)

Karsten1987 committed:
fix package.xml (#2)

wjwwood committed:
use updated rclcpp syntax for get_parameter_or (#4)

bulwahn committed:
make rostest in CMakeLists optional (ros/rosdistro#3010)

wjwwood committed:
update to support Python 2 and 3 (#559)

mikeferguson committed:
fix order of parameters (closes #553)

mikeferguson committed:
Merge pull request #565 from bulwahn/kinetic-devel

make rostest in CMakeLists optional (ros/rosdistro#3010)

mikeferguson committed:
Merge pull request #560 from wjwwood/map_server_fixup_cmake

fix cmake use of yaml-cpp and sdl / sdl-image in map_server

mikeferguson committed:
port #549 (in alphabetical order)

mryellow committed:
Added deps to amcl costmap_2d move_base (#512)

* [kinetic] amcl: Add std_srvs to find_package and catkin_package
* [kinetic] costmap_2d: Add visualization_msgs to find_package and catkin_package
* [kinetic] amcl: std_srvs only needed in find_package
* [kinetic] move_base: Add std_srvs to find_package

mikeferguson committed:
Merge pull request #551 from azaganidis/patch-1

Initialization of filter with GPS and odometry.

mikeferguson committed:
Merge pull request #567 from ros-planning/kinetic_553

fix order of parameters (closes #553)

codebot committed:
Add cost function to prevent unnecessary spinning

Adds a cost function and configuration parameter to penalize changes
in robot heading. This isn't relevant for diff-drive robots, where
heading changes are typically necessary to move towards goals.
However, this is useful for having control of the 'twirling' behavior
of holonomic robots. Setting the twirling_scale parameter to a high
value, say 50, causes the dwa planner to prefer to 'strafe' towards
goals rather than (eventually) aiming the robot at the goal while
driving. For some applications this is much preferred. This parameter
defaults to zero, which maintains the existing behavior unless it is
explicitly set to a non-zero value.

dhood committed:
Export map_server libraries (#7)

* Export libraries

* Use name of library

* ament_export_libraries takes list of strings

* Revert "ament_export_libraries takes list of strings"

This reverts commit 630312bf7d54e7ff584213b48da2340064de0450.

clalancette committed:
Switch use of rcl_system_time_now to rcutils_system_time_now. (#8)

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

mikaelarguedas committed:
use CMAKE_X_STANDARD and check compiler rather than platform

Commit:
amcl: fix compilation with gcc v7

When building amcl with recent enough gcc v7 compilation fails with
the error

src/amcl/map/map_cspace.cpp: In function 'void enqueue(map_t*, unsigned int, unsigned int, unsigned int, unsigned int, std::priority_queue<CellData>&, CachedDistanceMap*, unsigned char*)':
src/amcl/map/map_cspace.cpp:98:34: error: call of overloaded 'abs(unsigned int)' is ambiguous
   unsigned int di = abs(i - src_i);

Use `int abs(int)` flavour of the abs() function.

dhood committed:
Port AMCL to ROS 2 (and tf2) (#9)

* Remove dynamic reconfigure

* Remove rosbag functionality

* Remove message filter use

* tf -> tf2

* Strip leading / from frame IDs

* Re-enable message filter

* Don't use deperecated quaternion constructor

* Revert "Re-enable message filter"

This reverts commit c4160c2b8cb5925639c5fbe6b4d88ba120af07a9.

* Fix test failures by increasing transform timeouts

* WIP ros1->ros2

* Namespace of ::msg or ::srv

* ConstPtr -> ::ConstPtr

* ros::Time -> tf2::TimePoint

* ros::Duration -> tf2::Duration

* ros -> rclcpp

* ros2 timer

* Disable sigint handler

* Not using boost::shared_ptr

* Appropriate boost header for recursive_mutex

* Shared ptrs everywhere

* Use new durationFromSec tf2 function

* Subscribers using qos settings

* Leftover missing ::msg

* Use new tf2_ros::fromMsg

* Reinstate node ptr

* Disable most logging

* chrono fix

* 'msg.' -> 'msg->'

* Store node

* Create services properly

* Set parameters

* Comment more logging

* tf2 fixes

* fix service callback signatures

* tf2 changes

* misc

* only callbacks use shared_ptr

* map request service

* Fix boost link error

* Comment these for now just to move forward

* Use ament_auto_package

* tf2::Duration -> tf2::durationFromSec

* Re-enable "logging"

* Prevent node already added to executor crash

* (hack) use 'latest' TF info

* Debug

* Fixup (whitespace etc)

* Use rate for map request

* Don't use tf2 time for checkLaserReceived timer

* Clarify which TODOs are for before merge

* Add wait_for_service

* Restore initial qos choices

* Restore tolerance

* Demo ROS 1 launch file to be used with bridge

* [hack] Use map from topic instead of service

* Remove debug prints

* It probably won't be the map server publishing the map topic in that case (or we'd use the service)

* Expose use_map_topic as command line option

* Put initial pose back to 0 (can be set on 'initialpose' topic)

* Use tf2 time because ROS times can't be subtracted a.t.m.

* Revert test tolerance increase (needed for use without message filters)

* Remove commented delete that will never be re-enabled

* Tidy up workaround for 'latest' TF data

If the timestamp of the requested transform is 0 it will get latest

* Remove most workarounds for 'latest' TF data

* Prevent unused variable warnings

* Remove boost dependency

* Use CMAKE_CXX_STANDARD

* Alphabetical ordering

* Whitespace fixup

* Leftover debug prints

* Restore original default d_thresh

* Revert "Prevent unused variable warnings"

This reverts commit e624b1f45d03b8bdfedd23bee0c0d3bd7d5cd653.

* Prevent unused variable warnings

* Let ament_cmake_ros have control over the BUILD_SHARED_LIBS cmake flag

* Pedantic is a bit much

* simplify ament_target_dependencies call

* Correct usage of ament_target_dependencies

* Remove custom leading forward slash removal

Not necessary in ROS 2

* Remove ROS 1 testing launch file

* Avoid race condition with map_server topic usage

Issue with large services using fastrtps

nuclearsandwich committed:
Remove dependency on rostest. (#10)

dhood committed:
amcl integration: best effort QoS for laser scan subscription (#11)

* Make laser subscription best effort

Otherwise it can't subscribe to best effort publishers (and is generally more appropriate)

* Use rcutils logging macros

* Start a parameter server for external nodes to interact with

* Reduce diff with upstream

dhood committed:
Move map server executables to libexec directory (#12)

lengly committed:
fix typo

mikaelarguedas committed:
use new namespace (#13)

bponsler committed:
Exporting the map_server include directory (#14)

mikaelarguedas committed:
pcl proagate -lQt5::Widgets flag so we need to find_package Qt5Widgets (#578)

mikaelarguedas committed:
import only PCL common

mikaelarguedas committed:
add missing deps on libpcl

mikaelarguedas committed:
convert packages to format2

mikaelarguedas committed:
alphabetize dependency list

mikaelarguedas committed:
set message_generation build and runtime dependency

allenh1 commented:
@mikaelarguedas Do I run a packaging job for this one or the normal CI?

mikaelarguedas commented:
Can this be submitted upstream instead? looks like this overlap fixes already merged upstream (https://github.com/ros-planning/navigation/pull/587)

mikaelarguedas commented:
The tf2 stuff can be submitted in this repo given that ROS1 navigation port to tf2 is still pending.

> @mikaelarguedas Do I run a packaging job for this one or the normal CI?

Neither the packaging job or normal CI test this repository. Please launch a `ci_turtlebot-demo_linux` and `ci_turtlebot-demo_linux-aarch64`

allenh1 commented:
> Can this be submitted upstream instead? looks like this overlap fixes already merged upstream

Yes, that appears to be a similar patch.

Should I bother with a submission upstream since that fix was already merged? I could rebase on top of ros-planning/navigation, if that's the best solution (might be good to catch up anyway).

> Neither the packaging job or normal CI test this repository. Please launch a ci_turtlebot-demo_linux and ci_turtlebot-demo_linux-aarch64

Thanks.

mikaelarguedas commented:
> Should I bother with a submission upstream since that fix was already merged?

If the fix is similar no need to submit anything.

>  I could rebase on top of ros-planning/navigation, if that's the best solution (might be good to catch up anyway).

Yeah it would be great to catch up, that may be a mess to resolve conflicts. maybe it's worth waiting after https://github.com/ros-planning/navigation/pull/590 and https://github.com/ros-planning/navigation/pull/591 are merged

mikeferguson committed:
Merge pull request #590 from mikaelarguedas/remove_qt5_dependency

Cleanup PCL imports: fixes compilation on zesty and stretch

allenh1 commented:
> Yeah it would be great to catch up, that may be a mess to resolve conflicts. maybe it's worth waiting after ros-planning#590 and ros-planning#591 are merged

Definitely. I'll keep an eye on ros-planning#591.

mintar committed:
Fix CMakeLists + package.xmls (#548)

* Fix CMakeLists + package.xmls

This fixes compilation errors when rosjava is installed on the system.
It also removes a lot of errors reported by catkin_lint.

Fixes #537 .

* Fix more CMakeLists.txt + package.xmls

* amcl: disambiguate include path

This fixes the following catkin_lint warning:

    amcl: warning: include paths 'include/amcl/map' and 'include' are ambiguous
         * You have used two include paths where one is a parent of the
         * other. Thus the same headers can be included with two different
         * include paths which may confuse users. It is recommended that
         * you keep your include paths consistent.

wjwwood committed:
refactor to not use tf version 1 (#561)

mikeferguson committed:
Merge pull request #596 from ros-planning/lunar_548

Port #548 to lunar

corot committed:
Speedup (~60%) inflation layer update (#525)

* Replace the priority_queue used on inflation layer by a map containing cells to inflate split by distance from the closest obstacle

* Update inflation test to match previous change

mikeferguson committed:
don't update costs if inflation radius is zero

mikeferguson committed:
cleaner logic, fixes #156

mikaelarguedas committed:
rebase fixups

mikeferguson committed:
Merge pull request #600 from ros-planning/lunar_403

recompute cluster stat when force_publication

mikeferguson committed:
Merge pull request #598 from ros-planning/fix_156

cleaner logic, fixes #156

mikeferguson committed:
Merge pull request #591 from mikaelarguedas/package_format2

Convert to package format 2

mikaelarguedas commented:
@allenh1 FYI: Both referenced PRs have been merged

allenh1 commented:
@mikaelarguedas cool -- I'll rebase.

allenh1 commented:
@mikaelarguedas rebased on top of ros-planning/navigation:kinetic-devel and fixed the compiler warnings introduced by GCC 8.0.

Should I spin up a CI job?

mikaelarguedas commented:
Sorry for being unclear, the referenced PRs were targeting the lunar branch and not the kinetic-devel so we should rebase agianst the `lunar` branch.

Once rebased we should:
- review the changes (sanity checking at least not necessarily a thorough review )
- spin another round of CI
- test that the amcl demo still works
before merging.

allenh1 commented:
> Sorry for being unclear, the referenced PRs were targeting the lunar branch and not the kinetic-devel so we should rebase agianst the lunar branch.

Mop. That makes about 1000x more sense. I'll try again, lol.

allenh1 committed:
Merge branch 'ros2' into compile-for-gcc-7

allenh1 committed:
Fixed GCC 8.0 compiler warning.

allenh1 commented:
@mikaelarguedas Ok, rebased on top of the `lunar` branch. I'll launch a CI job after I finish rebuilding everything locally (started from a clean build just to make sure).

mikaelarguedas commented on map_server/CMakeLists.txt:
```diff
@@ -55,6 +55,7 @@ target_link_libraries(map_server
 )
 
 add_executable(map_server-map_saver src/map_saver.cpp)
+add_dependencies(map_server-map_saver ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
```
This is catkin specific. `catkin_EXPORTED_TARGETS` will not exist in an ament build

allenh1 replied:
Yup -- caught this one while building. My bad.

mikaelarguedas commented on map_server/src/image_loader.cpp:
```diff
@@ -42,6 +42,9 @@
 #include <SDL/SDL_image.h>
 
+// Use Bullet's Quaternion object to create one from Euler angles
+#include <LinearMath/btQuaternion.h>
```
bullet is not installed on the build machines AFAIK so this will likely fail to compile. We use tf2 quaternions in ROS 2 for now I think (see line 49)

allenh1 replied:
Ya, that's right. I'll fix it.

allenh1 committed:
Fix compilation, as well as remove more warnings.

allenh1 commented:
@mikaelarguedas Ok, all should be ready for CI now.

mikaelarguedas commented on map_server/src/main.cpp:
```diff
@@ -91,25 +91,25 @@ class MapServer
         try {
           doc["resolution"] >> res;
-        } catch (YAML::InvalidScalar) {
+        } catch (YAML::InvalidScalar &) {
```
These should be submitted upstream rather than here isn't it ? Or is it ROS 2 specific code?

allenh1 replied:
I suppose that should be upstreamed, yes. I'll do that right now.

allenh1 replied:
Ok, filed upstream.

mikaelarguedas replied:
can you put them in a separate commit in this PR ? That'll be cleaner for history and rebasing

allenh1 replied:
yes

allenh1 committed:
Fix GCC 8 warnings for map_server.

allenh1 commented:
@mikaelarguedas done

mikaelarguedas commented:
thanks ! looks good to me, feel free to launch CI on this. (note we wont "squash and merge" this one to keep history close to the one of the upstream repo)

allenh1 commented:
> thanks ! looks good to me, feel free to launch CI on this

will do

> (note we wont "squash and merge" this one to keep history close to the one of the upstream repo)

I was just about to ask... This is a "Rebase and Merge" then, right (assuming CI passes).

mikaelarguedas commented:
Not it will be a simple merge because we want the ros2 commits on top of the ros1 commits.
If we do "rebase and merge" we will rebase the ros1 commits on top of the ros2 commits.
(which is overall good news because rebase and merge seems to conflict :S)

allenh1 commented:
* Linux [![Build Status](http://ci.ros2.org/job/ci_turtlebot-demo_linux/84//badge/icon)](http://ci.ros2.org/job/ci_turtlebot-demo_linux/84/)
* Linux aarch64 [![Build Status](http://ci.ros2.org/job/ci_turtlebot-demo_linux-aarch64/92//badge/icon)](http://ci.ros2.org/job/ci_turtlebot-demo_linux-aarch64/92/)

allenh1 commented:
> Not it will be a simple merge because we want the ros2 commits on top of the ros1 commits.

Ok. Got it.

allenh1 commented:
Building and testing on a physical turtlebot 2, just to put this in the record.

allenh1 commented:
Ok, it works. Merging.
